src/core/server/saved_objects/migrations/integration_tests/cluster_routing_allocation_disabled.test.ts

```diff
@@ -0,0 +1,155 @@
+      },
+    },
+    {
+      oss: false,
+    }
+  );
+}
+const getClusterRoutingAllocations = (settings: Record<string, any>) => {
+  const routingAllocations =
+    settings?.transient?.['cluster.routing.allocation.enable'] ??
+    settings?.persistent?.['cluster.routing.allocation.enable'] ??
+    [];
+  return (
+    [...routingAllocations].length === 0 ||
+    [...routingAllocations].every((s: string) => s === 'all')
+  ); // if set, only allow 'all';
+};
+let esServer: kbnTestServer.TestElasticsearchUtils;
+
+async function updateRoutingAllocations(
+  esClient: ElasticsearchClient,
+  settingType: string = 'persistent',
+  value: string = 'none'
+) {
+  return await esClient.cluster.putSettings({
+    [settingType]: { cluster: { routing: { allocation: { enable: value } } } },
+  });
+}
+
+describe('unsupported_cluster_routing_allocation', () => {
+  let client: ElasticsearchClient;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+    esServer = await startES();
+    client = esServer.es.getClient();
+  });
+  afterAll(async () => {
+    await esServer.stop();
+  });
+
+  it('fails with a descriptive message when persistent replica allocation is not enabled', async () => {
+    const initialSettings = await client.cluster.getSettings({ flat_settings: true });
+
+    expect(getClusterRoutingAllocations(initialSettings)).toBe(true);
+
+    await updateRoutingAllocations(client, 'persistent', 'none');
+
+    const updatedSettings = await client.cluster.getSettings({ flat_settings: true });
+
+    expect(getClusterRoutingAllocations(updatedSettings)).toBe(false);
+
+    // now try to start Kibana
+    root = createKbnRoot();
+    await root.preboot();
+    await root.setup();
+
+    await expect(root.start()).rejects.toMatchInlineSnapshot(
+      `[Error: Unable to complete saved object migrations for the [.kibana] index: The elasticsearch cluster has cluster routing allocation incorrectly set for migrations to continue. To proceed, please remove the cluster routing allocation settings with PUT /_cluster/settings {"transient": {"cluster.routing.allocation.enable": null}, "persistent": {"cluster.routing.allocation.enable": null}}]`
+    );
+
+    await retryAsync(
+      async () => {
+        const logFileContent = await fs.readFile(logFilePath, 'utf-8');
+        const records = logFileContent
+          .split('\n')
+          .filter(Boolean)
+          .map((str) => JSON5.parse(str)) as LogRecord[];
+        expect(
+          records.find((rec) =>
+            rec.message.startsWith(
+              `Unable to complete saved object migrations for the [.kibana] index: The elasticsearch cluster has cluster routing allocation incorrectly set for migrations to continue.`
+            )
+          )
+        ).toBeDefined();
+      },
+      { retryAttempts: 10, retryDelayMs: 200 }
+    );
+  });
+
+  it('fails with a descriptive message when persistent replica allocation is set to "primaries"', async () => {
```

Comment:
I had added test cases for all the other unsupported settings but decided to only go with asserting against 'none' and 'primaries' replica allocations. The actions integration tests cover more.